[nextjs-react] Update to use @ syntax for remotes

JohnDaly opened the pull request:
## Changes
- Updating `nextjs-react` example to use the @ syntax for remotes
- Updating `nextjs-react` example to use `yarn` instead of `npm`. 
  - There was mixed usage of both in this example. I opted for `yarn` because it is used in more examples throughout the monorepo

## Other Fixes
- Adding `.npmrc` file to set privjs as registry for `@module-federation` packages
- Updating ESLint config in `nextjs-react/host-app` to fix linter warnings about `next/babel`
- Updating `webpack-cli` to fix `TypeError: cli.isMultipleCompiler is not a function` error
- Updating `webpack` and `webpack-dev-server` to latest versions

JohnDaly committed:
- Updating nextjs-react example to use yarn instead of npm
- Updating webpack, webpack-cli, and webpack-dev-server

JohnDaly committed:
Adding npmrc file to set privjs as registry for @module-federation packages

JohnDaly committed:
Updating ESLint config to fix linter warnings about next/babel

JohnDaly committed:
Updating example to use @ syntax for remotes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/module-federation/module-federation-examples?pullRequest=2067) <br/>All committers have signed the CLA.

JohnDaly commented on nextjs-react/remote-app/package.json:
```diff
@@ -8,9 +8,9 @@
-    "webpack-cli": "4.9.2",
-    "webpack-dev-server": "4.8.1"
+    "webpack": "5.73.0",
+    "webpack-cli": "4.10.0",
```
Fixes:

```
[0] [webpack-cli] TypeError: cli.isMultipleCompiler is not a function
[0]     at Command.<anonymous> (/module-federation-examples/nextjs-react/remote-app/node_modules/@webpack-cli/serve/lib/index.js:146:35)
[0]     at async Promise.all (index 1)
[0]     at async Command.<anonymous> (/module-federation-examples/nextjs-react/remote-app/node_modules/webpack-cli/lib/webpack-cli.js:1674:7)
```

JohnDaly commented on nextjs-react/host-app/.eslintrc.json:
```diff
@@ -1,3 +1,3 @@
 {
-  "extends": "next/core-web-vitals"
+  "extends": ["next/babel", "next/core-web-vitals"]
```
Fixes the `Parsing error: Cannot find module 'next/babel'` error from ESLint

ScriptedAlchemy reviewed (changes requested):
Please check the v4 release for nextjs-mf 

The api has changed slightly. No need to double configure MFP on next apps. All is done with the sidecar. Ideally I'd use the new major release if you're moving to the @ syntax

JohnDaly committed:
- Updating to v4 of @module-federation/nextjs-mf
- Updating host-app to use withFederatedSidecar

JohnDaly commented:
> Please check the v4 release for nextjs-mf
> 
> The api has changed slightly. No need to double configure MFP on next apps. All is done with the sidecar. Ideally I'd use the new major release if you're moving to the @ syntax

I've updated to the v4 release and updated the Next.js host app in the example to use `withFederatedSidecar` in https://github.com/module-federation/module-federation-examples/pull/2067/commits/ec84bdf38e1743a11fe20db03d0055ad29ee078d

ScriptedAlchemy commented:
Perfect. Will review tomorrow morning

JohnDaly committed:
Merge branch 'master' into fix/nextjs-react-update

JohnDaly committed:
yarn.lock update for remote-app

JohnDaly commented:
@ScriptedAlchemy do the changes look good to you?
